desktop/core/src/desktop/lib/idbroker/conf.py

```diff
@@ -13,61 +13,56 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from __future__ import absolute_import
-
 import logging
-import sys
 import requests
 
 from requests_kerberos import HTTPKerberosAuth
 from hadoop.core_site import get_conf
 
-if sys.version_info[0] > 2:
-  from django.utils.translation import gettext_lazy as _t
-else:
-  from django.utils.translation import ugettext_lazy as _t
+from django.utils.translation import gettext_lazy as _t
+
 
 LOG = logging.getLogger()
 
+
 _CNF_CAB_ADDRESS = 'fs.%s.ext.cab.address' # http://host:8444/gateway
 _CNF_CAB_ADDRESS_DT_PATH = 'fs.%s.ext.cab.dt.path' # dt
 _CNF_CAB_ADDRESS_PATH = 'fs.%s.ext.cab.path' # aws-cab
 _CNF_CAB_USERNAME = 'fs.%s.ext.cab.username' # when not using kerberos
 _CNF_CAB_PASSWORD = 'fs.%s.ext.cab.password'
+
 SUPPORTED_FS = {'s3a': 's3a', 'adl': 'azure', 'abfs': 'azure', 'azure': 'azure', 'gs': 'gs'}
 
 def validate_fs(fs=None):
   if fs in SUPPORTED_FS:
     return SUPPORTED_FS[fs]
   else:
-    LOG.warning('Selected FS %s is not supported by Hue IDBroker client' % fs)
+    LOG.warning('Selected filesystem %s is not supported by Hue IDBroker client.' % fs)
     return None
 
 def _handle_idbroker_ha(fs=None):
-  fs = validate_fs(fs)
-  idbrokeraddr = get_conf().get(_CNF_CAB_ADDRESS % fs) if fs else None
-  response = None
+  idbroker_addr_list = []
   if fs:
-    id_broker_addr = get_conf().get(_CNF_CAB_ADDRESS % fs)
-    if id_broker_addr:
-      id_broker_addr_list = id_broker_addr.split(',')
-      for id_broker_addr in id_broker_addr_list:
-        try:
-          response = requests.get(id_broker_addr.rstrip('/') + '/dt/knoxtoken/api/v1/token', auth=HTTPKerberosAuth(), verify=False)
-        except Exception as e:
-          if 'Name or service not known' in str(e):
-            LOG.warn('IDBroker %s is not available for use' % id_broker_addr)
-        # Check response for None and if response code is successful (200) or authentication needed (401)
-        if (response is not None) and (response.status_code in (200, 401)):
-          idbrokeraddr = id_broker_addr
-          break
-      return idbrokeraddr
-    else:
-      return idbrokeraddr
-  else:
-    return idbrokeraddr
+    idbroker_addr = get_conf().get(_CNF_CAB_ADDRESS % fs, '')
+    idbroker_addr_list = idbroker_addr.split(',')
+
+  response = None
+  for idb in idbroker_addr_list:
+    LOG.info('Attempting to connect to IDBroker URL: %s' % idb)
+
+    try:
+      response = requests.get(idb.rstrip('/') + '/dt/knoxtoken/api/v1/token', auth=HTTPKerberosAuth(), verify=False)
+    except Exception as e:
+      if 'Failed to establish a new connection' in str(e):
```

Comment:
Should we not handle Name or service not known as this could be a different one with failed to establish the connection.  Adding this information should also help

Reply:
I logged the response for both failure cases: When the whole host was down from AWS console and when the IDBroker instances were down from Data Lake CM.

In both cases, I didn't see `Name or service not known` string in response but `Failed to establish a new connection` was present both times. 

I think because of the former only old implementation was not working as expected and then I thought that why even check for it when its flaky and only depend on the stuff which was coming in response both times? I used `Failed to establish a new connection` in implementing other service HA also like Knox JWT and RAZ recently.

What do you think?